src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -368,9 +368,7 @@ public int compare(TypedVar o1, TypedVar o2) {
     // static methods. TS does not support static methods on interfaces, thus we create a new
     // namespace for them.
     if (isDefault && isInterfaceWithStatic(symbol.getType())) {
-      emitNamespaceBegin(symbol.getName());
-      treeWalker.walkDefaultInterface((FunctionType) symbol.getType());
-      emitNamespaceEnd();
+      declareNamespace(symbol.getName(), symbol, false, compiler, provides, isExtern);
```

Comment:
yey, recursion. You can add comment isDefault false guarantees termination, so that some further refactoring doesn't get us into infinite loops.